src/lsdb/catalog/catalog.py

```diff
@@ -1,20 +1,31 @@
 from __future__ import annotations
 
 import dataclasses
-from typing import Dict, List, Tuple, Type, cast
+import os
+from copy import copy
+from importlib.metadata import version
+from typing import Any, Dict, List, Tuple, Type, Union, cast
 
 import dask.dataframe as dd
 import hipscat as hc
+from hipscat.io import FilePointer, file_io
 from hipscat.pixel_math import HealpixPixel
+from typing_extensions import TypeAlias
 
 from lsdb.catalog.dataset.dataset import Dataset
 from lsdb.core.cone_search import cone_filter
 from lsdb.core.crossmatch.abstract_crossmatch_algorithm import AbstractCrossmatchAlgorithm
 from lsdb.core.crossmatch.crossmatch_algorithms import BuiltInCrossmatchAlgorithm
 from lsdb.dask.crossmatch_catalog_data import crossmatch_catalog_data
+from lsdb.io.file_io import write_dataframe_to_parquet
 
 DaskDFPixelMap = Dict[HealpixPixel, int]
 
+# Compute pixel map returns a tuple. The first element is
+# the number of data points within the HEALPix pixel, the
+# second element is the list of pixels it contains.
+HealpixInfo: TypeAlias = Tuple[int, List[int]]
```

Comment:
Or maybe not since I think this can be mostly removed when we take out `partition_info.csv`

Reply:
Yes, I had moved it here because it was causing a circular dependency. In fact I think it doesn't make sense to have these types scattered all over the place so I moved HealpixInfo and DaskDFPixelMap to a self-contained module!